### DISCUSSION TITLE
Neighbor list with type dependent cutoff radius

### DISCUSSION THREAD

**Discussion by LangLuke** [glotzerlab/hoomd-blue#1680] [Support] [answered]
Hi all,

I am running MD simulations with two particle types (A,B) such that B is much larger than A. This results in the pairwise interactions involving B having a much larger cutoff radius than the A-A interactions.

It seems excessive to store all A-A separations with the cutoff radius necessary for the B interactions. I was wondering if it was possible to use a neighbor list (or combination of neighbor lists) such that cutoff radius for A-B and B-B separations is large but the cutoff radius for A-A separations is small.

The documentation for the neighbor list structures state the following:
"The actual cutoff distance is always the maximum $r_{cut, i,j}$ of the base cutoff and associated pair potentials."
"Neighbor lists nominally include all particles within the chosen cutoff distances."

These two statements seem to imply what I am asking to do is impossible, but am curious if there is a workaround.

Thanks,
Luke

**Comment by mphoward** [comment, marked as answer]
The documentation is slightly unclear there, but the maximum is done per pair of types (i, j). So, the cutoff for the A-A interactions can indeed be much smaller than A-B or B-B, and HOOMD will handle this correctly out of the box.

I rewrote the neighbor list code several years ago because of exactly the issue and application you are describing, and I recommend using [`md.nlist.Tree`](https://hoomd-blue.readthedocs.io/en/v4.4.0/module-md-nlist.html#hoomd.md.nlist.Tree) for your neighbor list method. It was optimized for size asymmetric systems.

**Reply by joaander** [reply]
To clarify, the sentence "The actual cutoff distance is always the maximum $r_{cut, i,j}$ of the base cutoff and associated pair potentials" is trying to say that the r_cut_ij_overall = maximum(r_cut_ij_base, r_cut_ij_potential0, r_cut_ij, potential1, ...). In other words, the maximum is over potentials, not over i and j - the full matrix is kept: https://github.com/glotzerlab/hoomd-blue/blob/4bc6e20d15b83d2a1bf509351c217dc375f7a5e8/hoomd/md/NeighborList.cc#L463-L485

The "nominally includes" line is in the context of exclusions, where the documentation states how specifically excluded pairs will be removed from the neighbor list even when they are within a given r_cut distance.

**Reply by LangLuke** [reply]
Thanks!
